server/src/main/java/eu/fasten/server/plugins/kafka/FastenKafkaPlugin.java

```diff
@@ -142,16 +152,29 @@ private void handleConsuming() {
      * @param input input message [can be null]
      */
     private void handleProducing(String input) {
-        if (plugin.getPluginError() == null) {
+        try {
+            if (plugin.getPluginError() != null) {
+                throw plugin.getPluginError();
+            }
+
             var result = plugin.produce();
+            String payload = null;
+            if (result.isPresent()) {
+                if (writeDirectory != null && !writeDirectory.equals("")) {
```

Comment:
Better call `result.get()` before this `if` and store it in a variable. More debuggable and you can loose the `else`

Reply:
Made this part more debuggable and overall cleaned it a bit